streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -220,6 +220,13 @@ long sizeBytes() {
         long sizeInBytes = 0;
         for (final NamedCache namedCache : caches.values()) {
             sizeInBytes += namedCache.sizeInBytes();
+            if (isOverflowing(sizeInBytes)) {
+                return Long.MAX_VALUE;
+            }
+        }
+
+        if (isOverflowing(sizeInBytes)) {
```

Comment:
Do we need this check here?  Seems like we'll never hit this condition since we check after incrementing the size in the loop above and return if an overflow occurred.

Reply:
@bbejeck  Thanks for pointing out, infact i will fix the other mthid size() which has the same check which wont be reached/